components/camel-rabbitmq/src/main/java/org/apache/camel/component/rabbitmq/RabbitConsumer.java

```diff
@@ -56,9 +59,26 @@
             log.warn("Unable to open channel for RabbitMQConsumer. Continuing and will try again", e);
         }
     }
-
     @Override
     public void handleDelivery(String consumerTag, Envelope envelope, AMQP.BasicProperties properties, byte[] body) throws IOException {
+    	try {
+            if (!consumer.getEndpoint().isAutoAck()) {
+            	lock.acquire();
+            }
+            doHandleDelivery(consumerTag, envelope, properties, body);
```

Comment:
Maybe a `try` - `finally` block is more robust here? Just in case some strange exception comes out from `doHandleDelivery()`... I suggest the following snippet:

```Java
try {
    doHandleDelivery(consumerTag, envelope, properties, body);
} finally {
    if (!consumer.getEndpoint().isAutoAck()) {
        lock.release();
    }
}
```